BodyBuddy/ViewModels/IntakeViewModels/IntakeViewModel.cs

```diff
@@ -71,42 +77,80 @@ public async Task AddWaterClicked()
 		{
 			WaterCurrent += 250;
 			IntakeDetails.WaterCurrent = WaterCurrent;
+			WaterIntakeProgress = (double)WaterCurrent / (double)WaterGoal;
+			IntakeDetails.WaterProgress = WaterIntakeProgress;
 			await _intakeRepository.SaveChangesAsync(IntakeDetails);
 		}
 
 		[RelayCommand]
 		public async Task AddKcalClicked(int calories)
 		{
-			CaloriesCurrent += calories;
-			IntakeDetails.CalorieCurrent = CaloriesCurrent;
-			await _intakeRepository.SaveChangesAsync(IntakeDetails);
+			if(CaloriesCurrent + calories < 0)
+			{
+				ErrorMessage = "Cannot reduce current calorie intake below zero.";
+
+				CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+
+				ToastDuration duration = ToastDuration.Short;
+				double fontSize = 14;
+
+				var toast = Toast.Make(ErrorMessage, duration, fontSize);
+				await toast.Show(cancellationTokenSource.Token);
+			}
+			else
+			{
+				CaloriesCurrent += calories;
+				IntakeDetails.CalorieCurrent = CaloriesCurrent;
+				CalorieIntakeProgress = (double)CaloriesCurrent / (double)CalorieGoal;
+				IntakeDetails.CalorieProgress = CalorieIntakeProgress;
+				await _intakeRepository.SaveChangesAsync(IntakeDetails);
+			}
+		}
+
+		public string CalorieIntakeProgressAsPercentage
+		{
+			get { return $"{CalorieIntakeProgress * 100:F1} %"; }
 		}
 
 		#region popup methods
 
 		public async Task<bool> SaveNewIntakeGoal(string intakeType)
 		{
-			if (NewIntakeGoal == 0)
+			if (NewCurrentIntake < 0)
```

Comment:
Har NewCurrentIntake noget at gøre med ændring af goal?

Reply:
NewCurrentIntake og NewIntakeGoal bliver begge ændret i popuppen nu, så der kan ændres på ens current intake, hvis man er kommet til at taste for meget ind, derfor tjek på begge. Jeg giver metoden et nyt navn så den er mere beskrivende.